flipperdevices/flipperzero-firmware
# [FL-3764] Expansion module service improvements

# What's new

- Refactor the Expansion Module service to have a control thread and a worker thread
- Support for getting RX FIFO status in `furi_hal_serial` API (can use higher baud rates now)
- Set a different icon for the RPC session opened by an expansion module

# Verification 

1. Connect a supported expansion module and observe the new icon
2. Go to `Settings -> Expansion Modules` and try randomly changing settings both when the module is and is not attached. Flipper should not crash and remain operational at all times.
3. Go to `Settings -> System` and change the `Log Device` and `Log Baud Rate` under the same conditions as 2. Flipper should not crash and remain operational at all times.
4. Connect pin 13 to pin 16 and pin 14 to pin 15 and repeat steps described in 2 and 3.
5. Feed as much garbage input to the pin 13 as possible when expansion module support is enabled. Try connecting chatty 3rd-party modules that use USART and try running some apps. No crashes or functionality loss should occur.

# Checklist (For Reviewer)
